src/store/modules/app.js

```diff
@@ -161,6 +187,15 @@ const actions = {
       document.getElementsByTagName('html')[0].dataset.theme = '';
     }
     commit('setDarkmode', darkStatus);
+  },
+  loadApplicationState({ dispatch }) {
+    dispatch('toggleDarkMode', true);
+    dispatch('loadFavoriteKeyboard');
+  },
+  loadFavoriteKeyboard({ state, commit }) {
+    if (state.favoriteKeyboard) {
+      commit('setKeyboard', state.favoriteKeyboard);
```

Comment:
we should probably check if this is a valid keyboard and if not, delete it

Reply:
Valid keyboards means it is present in state.keyboards array?

Reply:
yeah, basically don't favorite something until after we have the full list of keyboards.